addons/sourcemod/scripting/superzombiefortress.sp

```diff
@@ -1662,81 +1709,53 @@ public Action Event_PlayerSpawn(Event event, const char[] name, bool dontBroadca
 				g_iInfectedCooldown[g_nInfected[iClient]] = iClient;			//client to prevent abuse to cycle through any infected
 			}
 			
-			if (g_nInfected[iClient] == Infected_Boomer
-				|| g_nInfected[iClient] == Infected_Charger)
+			if (TF2_GetPlayerClass(iClient) != GetInfectedClass(g_nInfected[iClient]))
 			{
-				if (TF2_GetPlayerClass(iClient) != TFClass_Heavy)
-				{
-					TF2_SetPlayerClass(iClient, TFClass_Heavy, true, true);
-					TF2_RespawnPlayer(iClient);
-					return Plugin_Stop;
-				}
-				
-				SetEntityRenderMode(iClient, RENDER_TRANSCOLOR);
-				
-				//Boomer
-				if (g_nInfected[iClient] == Infected_Boomer)
+				TF2_SetPlayerClass(iClient, GetInfectedClass(g_nInfected[iClient]), true, false);
+				TF2_RespawnPlayer(iClient);
+				return Plugin_Stop;
+			}
+			
+			switch (g_nInfected[iClient])
+			{
+				case Infected_Boomer:
 				{
+					SetEntityRenderMode(iClient, RENDER_TRANSCOLOR);
 					SetEntityRenderColor(iClient, 255, 255, 0, 255);
 					CPrintToChat(iClient, "{green}YOU ARE A BOOMER:\n{orange}- Call 'MEDIC!' to EXPLODE and JARATE nearby enemies!\n- You also explode upon dying, coating the killer and assister in JARATE.");
```

Comment:
All of those can go to config, all special infected uses it (excluding tank with `CPrintToChat`)